### PR TITLE
feat(fxa-settings): add a simple progress bar for the recovery flow

### PR DESCRIPTION
**Pull request by millsoper**
## Because:

* We need a simple progress bar to track the user's progress through the account recovery flow

## This commit:

* Adds in a simple progress bar

Closes #FXA-7229

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)
Firefox:
Screen Shot 2023-04-19 at 4 12 56 PM
Safari:
Screen Shot 2023-04-19 at 4 13 04 PM
Chrome:

### CODE REVIEW

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.test.tsx
Comment (pdehaan) [inline packages/fxa-settings/src/components/Settings/ProgressBar/index.test.tsx:9]: nit: should this be "ProgressBar"?
"Security" seems a bit... misleading.

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx
Comment (pdehaan) [inline packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx:30]: I tried checking these in the [Storybook build](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/101e87cca72e9d41f403bd4cc006fb5a3ab24fb0/fxa-settings/index.html?path=/story/components-settings-progressbar--second-of-five-steps), but the progress bars were both 0%.
FinalStep below was 100%.

```html
<!-- FirstOfFiveSteps -->
<div aria-label="A progress bar showing that the user is on step 1 of 5." role="progressbar" aria-valuemin="1" aria-valuemax="5" aria-valuetext="1" class="h-2 w-full bg-grey-100 rounded flex ltr:flex-row rtl:flex-row-reverse"><div class="h-2 w-1/5 bg-blue-400 rounded"></div></div>

<!-- SecondOfFiveSteps -->
<div aria-label="A progress bar showing that the user is on step 2 of 5." role="progressbar" aria-valuemin="1" aria-valuemax="5" aria-valuetext="2" class="h-2 w-full bg-grey-100 rounded flex ltr:flex-row rtl:flex-row-reverse"><div class="h-2 w-2/5 bg-blue-400 rounded"></div></div>

<!-- Final Step -->
<div aria-label="A progress bar showing that the user is on step 5 of 5." role="progressbar" aria-valuemin="1" aria-valuemax="5" aria-valuetext="5" class="h-2 w-full bg-grey-100 rounded flex ltr:flex-row rtl:flex-row-reverse"><div class="h-2 w-full bg-blue-400 rounded"></div></div>
```

But per the [Tailwind `width` docs](https://tailwindcss.com/docs/width), everything seems right (unless this is a z-index issue and one is just obscuring the other).

In the Tailwind I see widths like `w-1/2` and `w-2/3` and `w-3/4` and `w-4/5` and `w-5/6` then we jump to `w-11/12`. So there _might_ be a potential for some edge cases if we have an unholy wizard that is between 7 and 11 steps.

Our markup/classes looks close to https://flowbite.com/docs/components/progress/ implementation. 🤔

Comment (pdehaan) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx:30]: Screenshot 2023-04-18 at 11 11 24 AM

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.test.tsx
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.test.tsx:9]: oh god it is that's a typo sorry

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx:30]: huh....I thought I had fixed this. Looking into it again! Also grateful to see the words "unholy wizard" in a PR

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx:30]: It was for custom styling but let me see what I can do! It would be really nice to go with the real semantic one...

Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx:31]: Yeah, the styling seems to be quite hard to overwrite, and the accessibility is relatively simple to wire up, so I think I'm going to lean away from using the real `<progress>` element, much as it pains me :(

File: packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl
Comment (bcolsson) [inline packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl:7]: I'm curious, is there a style for aria labels?
I see many of the strings use `aria-label` as an attribute, example:
https://github.com/mozilla/fxa/blob/def80f81d48ecbfbf54b7e0ef487a8a2c753771e/packages/fxa-settings/src/components/images/en.ftl

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx:30]: Would it be possible to use a percentage as an arbitrary value? 🤔 
`w-[${currentStep/numberOfSteps}%]`

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx
Comment (vpomerleau) [inline packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx:32]: Nice work on adding all the accessibility attributes for the progressbar!! I learned something new today :)

I wasn't sure if the aria-label would get picked up because this is a non-interactive element, but Deque indicates its good 👍 🎉 
https://dequeuniversity.com/rules/axe/4.1/aria-progressbar-name

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx
Comment (pdehaan) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx:30]: > Also grateful to see the words "unholy wizard" in a PR

/Clippy has entered the chat! 📎 

---

Re: arbitrary values,

https://tailwindcss.com/docs/width#arbitrary-values sayeth:
> If you need to use a one-off `width` value that doesn’t make sense to include in your theme, use square brackets to generate a property on the fly using any arbitrary value.
>
> ```html
> <div class="w-[32rem]">
>  
> </div>
> ```

So wonder if we could do something like `w-[46%]`.
But I'm also happy to completely ignore me and be satisfied with the knowledge that we'll never have a wizard that is more than 6 steps.
You could even code that into the component to throw a fatal error if `numberOfSteps` > 6.

But still curious if switching to `<progress>` would solve some of these and give potentially better a11y support.

File: packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl
Comment (pdehaan) [reply packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl:7]: I'll defer to wiser minds, but I think `aria-label` ([MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)) is only used for screen readers and is not meant to be part of the UI.
> _**Note:** The `aria-label` is only "visible" to assistive technologies. If the information is important enough to add for AT users, consider making it visible for all users._

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx
Comment (pdehaan) [inline packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx:24]: Nit: you're pointing to legacy tailwindcss@2 docs, but we're currently on ^3.2.0.
If I change the URL to omit the "v2." subdomain, that page doesn't seem to exist in the v3 world, but this might be closest: https://tailwindcss.com/docs/width#using-custom-values

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx:30]: OH somehow it had not sunk in for me that arbitrary values could include percentages??? I will test. I wish the styles for the `<progress>` element weren't so opinionated :/. The combination of how specific those styles are, plus the fact that we're using tailwind makes it a little picky. Altho....I suppose I could just write my own custom class with raw CSS inside. I dropped the idea because tailwind itself just recommended making a progress bar from a combination of divs, so it seemed like a widely accepted standard.

File: packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl:7]: I've been confused about that label historically but since there is a precedent, I'm more than happy to add it!

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx:24]: Ohhhhh thank you Peter!

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.stories.tsx:30]: Okay I had some issues with the percentages, BUT I successfully got the `<progress>` bar styled for webkit-based and moz browsers!

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx
Comment (millsoper) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx]: FIXED!

File: packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl
Comment (bcolsson) [reply packages/fxa-settings/src/components/Settings/ProgressBar/en.ftl:7]: I'll leave the call on what to do up to those with more knowledge, just something I noticed but doesn't really affect localization.

**Comment by vpomerleau** [comment]
Sweet! Looks great!! 🎉 I've also checked it out in Windows HCM with Storybook, and it's now visible, hurray! It's showing up with original color (blue) which is different from the Figma design, but that's probably fine because the yellow is meant to indicate interactive link elements (and the progress bar isn't).

File: packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx
Comment (vpomerleau) [reply packages/fxa-settings/src/components/Settings/ProgressBar/index.tsx:31]: @millsoper 
We might still need to add `role='progressbar'` for JAWS screen reader, but an aria-label might not be needed with the semantic `progress` element according to this: 
https://accessible-app.com/pattern/vue/progress
Not too sure about the focus management, to ensure that the steps are actually read out 🤔 

(All this complexity could be avoided if we got the green light from UX 🟢 to add "Step 1  of 4" back into the UI 😉 )

**Review by vpomerleau — APPROVED** [review]

**Pull request merged**
